test-data/unit/lib-stub/dataclasses.pyi

```diff
@@ -0,0 +1,36 @@
+from typing import Any, Callable, Optional, TypeVar, overload
+
+_T = TypeVar('_T')
+_C = TypeVar('_C', bound=type)
+
+
+@overload
+def dataclass(_cls: _C,
```

Comment:
You also need to add `InitVar` (here and on typeshed). You can define it as a normal generic class:
```python
class InitVar(Generic[_T]):
    pass
```
Then in the plugin you can test for it e.g. like this `if isinstance(some_type, Instance) and some_type.info.fullname() == 'dataclasses.InitVar'`.